tests/xharness/Jenkins/TestTasks/TestTask.cs

```diff
@@ -32,6 +33,16 @@ public abstract class TestTask
 		public bool BuildOnly;
 		public string KnownFailure;
 
+		public TestTask ()
+		{
+			// no, Interlocked.Increment is not going to do the samething, it is a two step, set and increment, 
```

Comment:
I don't understand why using Interlocked.Increment wouldn't work:

```csharp
ID = Interlocked.Increment (ref counter);
```

Reply:
I could have read the lock wrong. But I don't like locks in constructors and are a bad practice. I move to Increment and if we see it happening again we know where is the issue.

Reply:
@rolfbjarne fixed.